### ISSUE TITLE
Improve dependency on drracket package

### ISSUE THREAD

**Issue by greghendershott** [greghendershott/racket-mode#471]
Arising from #468:

Currently we recommend `raco pkg install drracket`. This ensures we have deps like:

- ` drracket-tool-lib`
-  `errortrace-lib`
- `macro-debugger-text-lib`
- `rackunit-lib`
- `html`
- More?? 

Also:

- `gui-debugger/marks` is only supplied by the full `drracket` pkg, but, it would be very easy to "vendor" that as we're already doing for the annotator to fix a couple issues.

What to do:

- [X] Determine the _complete_ minimal set of actual deps. This is somewhat complicated by Racket Mode's use of `lazy-require` to defer loading some of its own command code -- but that also defeats using `raco setup --check-pkg-deps` to determine the list of dependencies. For example I used that to discover some of the deps, but it didn't report `html`. I only discovered that by using Racket Mode on a 7.7-minimal install, and eventually used some command that transitively needed `html`.

- [X] Document that, including the `raco pkg install -auto x y z` command to run.

- [X] Consider an Emacs Lisp function/command to check the deps are installed, and if not, offer to run that command.

**Comment by greghendershott** [comment]
I'm going to close this because https://www.racket-mode.com/#Minimal-Racket does now document the minimal set of packages needed, if one does _not_ want to use `racket-xp-mode`. Otherwise, `drracket-tool-lib` is also needed -- but that pulls in nearly all of what `drracket` does (partly due to `racket-index` depending on `gui-lib`) so people may as well just go with `drracket` (which they get already with non-Minimal Racket).

As for an Emacs command to check deps and run `raco pkg install`: I think this is messy because of the choice just discussed (there is no "the" deps, no one set right for everyone), multiplied by the fact that we now support multiple back ends, both local and remote.

TL;DR the first 2 items are done, and the 3rd was to "consider" a command -- which was considered, but not done.

---

Someday if things like `racket-index` can change such that `drracket-tool-lib` no longer transitively requires `gui-lib`, this would be worth revisiting. In that case, at least the docs could be updated. Maybe then we could just recommend one set of deps (if `drracket-tool-lib` is light enough, just recommend it even if someone won't use `racket-xp-mode`). And if it's just the one set of deps, maybe the Emacs command etc. make more sense after all.

**Comment by greghendershott** [comment]
Re-opening because I need to update some explanations:

1. There is now a `drracket-tool-text-lib` which does _not_ depend on `gui-lib`. :tada:

2. Unfortunately, `racket-index` _does_ still depend on `gui-lib` (IIRC a build dep for `gui-lib` identifiers in doc links). :cry:

So it is `racket-index` that would needs its deps lightened. There was some discussion on GitHub or Slack about this, I need to find. I'm willing to help with that, if I'm able.

**Comment by greghendershott** [comment]
https://github.com/racket/racket/blob/master/pkgs/racket-doc/info.rkt#L18
